File: frontend/src/types/analysis.ts
export type AnalyzePromiseRequest = {
  text: string;
};

export type Criterion = {
  key: string;
  name: string;
  weight: number;
  status: "yes" | "partial" | "no";
  score: number;
  explanation: string;
};

export type PublicSource = {
  name: string;
  description: string;
};

export type Evidence = {
  source: string;
  title: string;
  description: string;
  url: string;
  indicator: string;
  value: number;
  unit: string;
  reference: string;
};

export type AnalyzePromiseResponse = {
  summary: string;
  score: number;
  confidence: number;
  target_value: number;
  target_unit: string;
  criteria: Criterion[];
  risks: string[];
  sources: PublicSource[];
  evidence: Evidence[];
};

File: backend/internal/domain/analysis.go
package domain

type CriterionStatus string

const (
	CriterionStatusYes     CriterionStatus = "yes"
	CriterionStatusPartial CriterionStatus = "partial"
	CriterionStatusNo      CriterionStatus = "no"
)

type Criterion struct {
	Key         string
	Name        string
	Weight      int
	Status      CriterionStatus
	Score       float64
	Explanation string
}

type PublicSource struct {
	Name        string
	Description string
}

type Evidence struct {
	Source      string
	Title       string
	Description string
	URL         string
}
type Analysis struct {
	Summary    string
	Score      int
	Confidence int
	Criteria   []Criterion
	Risks      []string
	Sources    []PublicSource
	Evidence []Evidence
}
